impress-org/givewp-next-gen
# Feature: Add auto-generated, unique field names

Resolves #24

## Description



This PR adds support for field names that are unique and predictable, auto-generating incremented suggestions when a duplicate is found, and originally derived from the field's label.

## Todo

- [x] Validate for unique
- [x] Validate for required
- [x] Validate for format (slug)
- [x] Update Block to Fields API conversion to support field names

## Visuals

![image](https://user-images.githubusercontent.com/10858303/193310760-8b97e20e-76f6-4581-8a3f-b2e9454b2cca.png)

## Testing Instructions



- Create a new, custom Text Field and set the Field Name attribute to `tester`.
- Create a another new, custom Text Field and set the Field Name attribute to `tester`.
  - The Field Name attribute should be updated to `tester-1` on blur.
- Create a another new, custom Text Field and set the Field Name attribute to `tester`.
  - The Field Name attribute should be updated to `tester-2` on blur.

## Pre-review Checklist
